doobie-hikari/src/main/scala/com/avast/sst/doobie/DoobieHikariModule.scala

```diff
@@ -0,0 +1,76 @@
+package com.avast.sst.doobie
+
+import java.util.concurrent.{ScheduledExecutorService, ThreadFactory}
+
+import cats.Show
+import cats.effect.{Async, Blocker, ContextShift, Resource, Sync}
+import cats.syntax.show._
+import com.zaxxer.hikari.HikariConfig
+import com.zaxxer.hikari.metrics.MetricsTrackerFactory
+import doobie.enum.TransactionIsolation
+import doobie.hikari.HikariTransactor
+
+import scala.concurrent.ExecutionContext
+
+object DoobieHikariModule {
+
+  /** Makes [[doobie.hikari.HikariTransactor]] initialized with the given config.
+    *
+    * @param boundedConnectExecutionContext [[scala.concurrent.ExecutionContext]] used for creating connections (should be bounded!)
+    */
+  def make[F[_]: Async](
+    config: DoobieHikariConfig,
+    boundedConnectExecutionContext: ExecutionContext,
+    blocker: Blocker,
+    metricsTrackerFactory: Option[MetricsTrackerFactory] = None
+  )(implicit cs: ContextShift[F]): Resource[F, HikariTransactor[F]] = {
```

Comment:
maybe we should return `Transactor.Aux[M, DataSource]` (instead of `HikariTransactor[F]` == `Transactor.Aux[M, HikariDataSource]`)
:thinking:

Reply:
I don't think so. You can always "upcast" it to that but you can never "downcast" it. I think the "rule" is to accept the most generic thing and return the most specific thing. Returning `HikariTransactor` allows the user to use something specific about it (just in case they need it).

Reply:
yeah, that's totally my question here actually: do we want the downstream users to "know" this is not a Transactor on just any `DataSource`, but it is on `HikariDataSource` specifically?

maybe that's an implementation detail we want to abstract away :smirk:

reading from the sources, the Hikari specific members are: `HikariPoolMXBean getHikariPoolMXBean()`, `HikariConfigMXBean getHikariConfigMXBean()` and `void evictConnection(Connection connection)`

Reply:
I do think we don't want to abstract it. Initialization modules are not an abstraction, they just help you initialize and put stuff together. I think you want to get the specific type because what you will do you will use it as `Transactor[F]` anyway but just in case you needed to call one of those methods you have the option in your `main` method. I think this is really good and unharmful strategy that can be applied without thinking.